fix: improve UserWarning message (#7685)
PyTorchLightning/pytorch-lightning: CHANGELOG.md

### Fixed

> - Fixed a potential confusion about which of overfit or shuffle is being disabled in the UserWarning message when both overfit and training dataloader shuffling are enabled ([#7685](https://github.com/PyTorchLightning/pytorch-lightning/pull/7685))

Reviewer: ```suggestion
- Fixed ambiguous warning when both overfit and train dataloader shuffling are enabled ([#7685](https://github.com/PyTorchLightning/pytorch-lightning/pull/7685))
```
